fdm-data/src/cultivations/hash.ts

```
@@ -0,0 +1,25 @@
+    cultivation.hash = null
+
+    // Remove all keys without a value
+    const filteredCultivation = Object.fromEntries(
+        Object.entries(cultivation).filter(
+            ([, value]) => value !== undefined || value !== null,
+        ),
+    )
+
+    // Sort keys to ensure consistent hash generation for identical objects
+    const sortedKeys = Object.keys(filteredCultivation).sort()
+    const sortedCultivation = sortedKeys.reduce((obj: any, key) => {
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)